app/controllers/users_controller.rb

```diff
@@ -1,18 +1,53 @@
 class UsersController < ApplicationController
-  before_action :set_user
+  before_action :set_user, except: [:index, :new, :create]
+  after_action :verify_authorized, except: :index
+  after_action :verify_policy_scoped, only: :index
+
+  def index
+    @users = policy_scope(User)
```

Comment:
I might not have made myself clear. There is nothing wrong in this controller/method. I was referring to `Api::V1::UsersController#index`. The issue I was raising there is regarding the conditional structure of setting a scope, filtering it according to params and having the return be a scope called on the filtered value. That can lead to surprising results for API calls, as we've seen in the past.